test/unit.html

```
@@ -54,35 +49,130 @@
     <h1>Unit Tests</h1>
     <div id="mocha"></div>
     <div id="mock"></div>
+    <div id="qunit"></div>
 
     <script type="text/javascript">
       teardown(function() { $('#mock').empty(); });
+      var json = location.search.indexOf('json') >= 0;
+      var listTests = location.search.indexOf('listTests') >= 0;
+      var startTime = Date.now();
+      var moduleResults = [];
+      var currentSuite;
       var runner = mocha.run();
 
-      if (post_xunit_to) {
-        // the following is based on https://github.com/saucelabs-sample-scripts/JavaScript/blob/4946c5cf0ab7325dce5562881dba7c28e30989e5/reporting_mocha.js
-        var failedTests = [];
-        runner.on('fail', function(test, err) {
-          function flattenTitles(test) {
-            var titles = [];
-            while (test.parent.title) {
-              titles.push(test.parent.title);
-              test = test.parent;
-            }
-            return titles.reverse();
-          }
+      runner.on('suite', function(suite) {
```

I think the big difference between TestDriver and this is that we are actually going to run the tests during the listTests phase and just pretend they all pass rather than actually mocking the tests out to make sure we don't have failures in the tests prevent us from getting the full list of tests.

I don't know what a "suite" is but we should be careful to make sure that all "suites" are reported during `listTest`.  The thing I'm worried about is maybe a suite doesn't run because there was already a failure reported. As long as suites never depend on other suites passing I think this is good to go.